model/implementation/DeliveryService.php

```diff
@@ -610,23 +647,36 @@ public function resumeSession(AssessmentTestSession $session)
     }
 
     /**
-     * Returns a group assinged to the delivery
+     * Returns a group assigned to the delivery
      *
      * @param string $deliveryId
-     * @return string
+     * @param string $testCenterId
+     * @return Resource
      */
-    private function findGroup($deliveryId)
+    private function findGroup($deliveryId, $testCenterId)
     {
-        $groups = GroupsService::singleton()->getRootClass()->searchInstances(array(
-            PROPERTY_GROUP_DELVIERY => $deliveryId
-        ), array(
+        $groups = $this->getGroupClass()->searchInstances(
+            array(
+                PROPERTY_GROUP_DELVIERY => $deliveryId,
+                self::PROPERTY_GROUP_TEST_CENTERS => $testCenterId
+            ),
+            array(
             'recursive' => true, 'like' => false
         ));
         if (empty($groups)) {
-            \common_Logger::w('No system group exists for delivery '.$deliveryId.'. creating one');
-            $delivery = new \core_kernel_classes_Resource($deliveryId);
-            $newGroup = GroupsService::singleton()->getRootClass()->createInstance('test takers for delivery '.$delivery->getLabel());
-            $newGroup->setPropertyValue(new \core_kernel_classes_Property(PROPERTY_GROUP_DELVIERY), $deliveryId);
+            \common_Logger::w('No system group exists for delivery '.$deliveryId.' and test center '.$testCenterId.'. creating one');
+            $delivery = new Resource($deliveryId);
+            $testCenter = new Resource($testCenterId);
+            $instanceName = 'test takers for delivery '.$delivery->getLabel().' and Test center '.$testCenter->getLabel();
```

Comment:
Not very important, but why is `Test center` in upper case while `test takers` and `delivery` are not ?